PowerPointLabs/PowerPointLabs/ThisAddIn.cs

```diff
@@ -1022,7 +1052,19 @@ private void ThisAddInPresentationOpen(PowerPoint.Presentation pres)
 
         private void ThisAddInPresentationClose(PowerPoint.Presentation pres)
         {
-            Trace.TraceInformation("Closing " + pres.Name);
+            Trace.TraceInformation("Closing " + pres.Name + "...");
+
+
+            // We need to check if there is only one window AND the active window's presentation 
+            // has the same name as the one we are closing because it is possible for background 
+            // presentation (those without windows) to close and trigger this event as well.
+            // We only want to shut down PPTLabs if we are closing the main presentation.
+            if (Application.Windows.Count == 1 &&
+                Application.ActiveWindow.Presentation.FullName == pres.FullName)
+            {
+                // If this current window we are closing is the last window, then PptLabs should terminate.
+                _pptLabsShouldTerminate = true;
+            }
 
             if (IsApplicationVersion2010() &&

```

Comment:
Since there is this part of the code only for PPT 2010, we definitely need to test this PR with PPT 2010. And is this part of the code still needed? It seems to be shutting down shapes lab presentation only if presentations.count == 2

Reply:
My bad... For some reason i seemed to have missed out this part of the code :P. I'll have to investigate 2010 a little more...

Reply:
This code section has been removed. Does not affect PPT2010, windows and presentations still close as they should.